### ISSUE TITLE
Potentially out of date

### ISSUE THREAD

**Issue by quozl** [cristian99garcia/periodic-table#14]
The most recent discoveries do not appear to have been integrated; based on the date of last change to the table upstream and the commits to the code here in Sugar Labs.  Needs careful checking.  A script to do the check may be in the repository already, I've not checked.

https://en.wikipedia.org/wiki/Periodic_table

**Comment by quozl** [comment]
In addition, the data/*.pdf files have not been changed for a long time.

**Comment by S-kus** [comment]
Hey @quozl,
I would like to work on this issue,
I have a doubt, Do I need to install the activity on my Sugar desktop to test, and then accordingly after changing the file I should raise the pull request in this directory (I can do this if required), or is there any other way to run just this activity without Sugar (by running only this repository).

Currently, for a particular element _.pdf_ is rendering, Now if anyone wants to update the pdf by changing the data, It's difficult to do in the same format (the way is already adjusted i.e. side block, etc).
**My suggestion** for this is that we can create a _.json_ file or _.md_ file for the element data and then a component to render these data(in the frontend) in a suitable way. This will help if any further updates are required to be done in the data in the future of the element(as less code change will be there), and also not affect the UI of the final activity.

**Comment by quozl** [comment]
Thanks for offering to work on this issue.

Yes, you should test your changes, otherwise when you make a pull request and say that it is untested the pull request will wait for someone else to have the time to test, and then if there is a regression it will have to be communicated back to you.  All this takes extra time.  However, if your changes are only to the PDF files or the embedded data, the chances of a regression are much lower, so we might be able to argue to reduce testing.

To test in Sugar, clone your repository to a new directory `~/Activities/PeriodicTable.activity` and then restart Sugar.  This will list the activity on Sugar Home View.  You can make changes to the activity (by editing or `git pull` from your branch) and then stop and start the activity to test the change.

Problem with switching to another format is how to verify the pull request is valid.  You should find out how the PDF files were made.  They appear to be Wikipedia pages saved as PDF.  Perhaps the tool used is still available, or another tool is available.  It would be good to document the process in this repository.

The upstream data source https://github.com/Bowserinator/Periodic-Table-JSON/blob/master/PeriodicTableJSON.json also has had commits since it was copied into the activity.  These should be taken into the activity.

See also
* https://wiki.laptop.org/go/Wikislices
* https://wiki.laptop.org/go/Chemistry

**Comment by S-kus** [comment]
For pdf files, I am working to find a tool to make the pdf in the same manner with images as it is currently.


> The upstream data source https://github.com/Bowserinator/Periodic-Table-JSON/blob/master/PeriodicTableJSON.json also has had commits since it was copied into the activity. These should be taken into the activity.

Sure, I will update this too.

> See also
> 
> * https://wiki.laptop.org/go/Wikislices
> * https://wiki.laptop.org/go/Chemistry

I go through the Wikislices periodic table activity in Chemistry action, They are rendering .html files and no images are there like in Sugar. 

Till now, what I found I can use the edit page feature of the Wikipedia page, and then after editing it accordingly I will download it and remove the upper extra dates and other irrelevant things by cropping or editing the pdf manually.

**Comment by quozl** [comment]
Have you tried wiki2pdf as described in those pages?

**Comment by S-kus** [comment]
Yes, @quozl, I go through pdf2pdf as well but I don't know how to get those wiki files and edit them. 
If I follow the [edit page](https://en.wikipedia.org/wiki/Nobelium?action=edit&veswitched=1) of an element on Wikipedia or it's [raw page](https://en.wikipedia.org/wiki/Nobelium?action=raw), and then removed the required content but I am still not able to get how to save them (with what file extension) and then proceed by converting them to pdf.

**Comment by quozl** [comment]
Okay, that's unfortunate.  There's little point more than one of us learning how to do it.  You can give up if you like.  If I were to try it (not planning to), the next step might be to test the _mediawiki2latexguipyqt_ package on Ubuntu, then script the underlying _mediawiki2latex_ package.  The package description says it generates LaTeX or PDF.

**Comment by S-kus** [comment]
@quozl, I will update here if I find an effective way of doing it. 
Thanks for this package information on Ubuntu, I will try to explore it.

**Comment by S-kus** [comment]
Hey, @quozl I had contacted @cvgarciarea about it, as he is the one who did this earlier. Thanks to him. I tried to do it in the same way. 
Please review  [this pdf](https://drive.google.com/file/d/1wR1HEuEBF5sjiYhMhPYlxGXbviQRTy_-/view?usp=sharing). It's an updated one from Wikipedia. 
What's your view about it?
If it's fine, Can I update other data in a similar manner?

**Comment by quozl** [comment]
Thanks.  Reviewed;

* [x] many words have underlines; and the underlines are not linked text,
* [x] no footnotes section,
* [x] no bibliography section,
* [ ] very different font, text size, and aspect ratio of page,

Original design of the app did not access network on first click, and the `data/102.pdf` file has only one link to the Wikipedia page at the end.

Compare with Wikipedia's own PDF maker;
* https://en.wikipedia.org/wiki/Help:Download_as_PDF
* https://www.mediawiki.org/wiki/Proton

The result has coloured element table, a table of contents, and other features.

I'd like to hear other opinions.

**Comment by S-kus** [comment]
@quozl I updated the pdf with the required changes in [this link.](https://drive.google.com/file/d/1SYssJuVUe-l5pt4UhExaHIRDhrpQgN6A/view?usp=sharing)
> 
> * [ ]  many words have underlines; and the underlines are not linked text,

Now it's corrected.

> * [ ]  no footnotes section, no bibliography section

It's not in the current `data/102.pdf` so I thought this might not be needed. It will also increase the file length and if anyone wants to refer one can easily follow the Wikipedia link at the bottom.

> * [ ]  very different font, text size, and aspect ratio of page,

I corrected the aspect ratio. If I talk about font, could you please confirm which `font-family` we need here, I will update it via dev tools.


> The result has coloured element table, a table of contents, and other features.

I added color and made the data similar to the current pdf.

Could you please review it once more and let me know if any other changes are required?

**Comment by quozl** [comment]
Thanks.  Reviewed.  I don't know what font is used, but you can probably find out by reading the existing PDFs.

**Comment by S-kus** [comment]
Using this [online tool](https://www.artworkflowhq.com/artwork-proofing/font-finder), I detect it's using `font-family: DejaVuSans`.
I am trying to edit the updated pdf, as this font is not rendering via dev tools.

**Comment by S-kus** [comment]
@quozl on going through the repository and also using [this tool](https://www.artworkflowhq.com/artwork-proofing/font-finder) I found out it's using `Deja Vu sans` or `Deja Vu` font family. 

So, I updated the pdf with the correct font and now this updated pdf is similar to 'data/102.pdf'. 
If it's fine then can I raise the pr to update the data? 
Also 


> It would be good to document the process in this repository.

do this too in the documentation it would be really helpful for future updates.

**Comment by quozl** [comment]
Guess so.

**Comment by S-kus** [comment]
Sorry, by mistake I didn't share the updated pdf (with the font).
[This is the pdf link.](https://drive.google.com/file/d/1y_ZJi3g-JzBx0WpSts74Nca7rWLZxkAZ/view?usp=sharing)

So, I will raise the pull request with documentation and data's pdf.

**Comment by quozl** [comment]
Thanks.
